modules/standard/CTypes.chpl

```diff
@@ -0,0 +1,726 @@
+/*
+ * Copyright 2020-2022 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// CTypes.chpl
+//
+
+/*
+
+Defines C types and related routines to support interoperability
+
+This module provides access to common C types for the purpose of
+calling between Chapel and C, declaring variables using C's types,
+etc.  It also contains routines in support of working with C types.
+
+See :ref:`readme-extern`, and :ref:`Chapter-Interoperability` for
+additional information about interoperating between Chapel and C (or
+other languages).
+
+.. include:: ChapelSysCTypes.rst
+
+*/
+
+module CTypes {
+  use HaltWrappers;
+  public use ChapelSysCTypes;
+
+  pragma "no doc"
+  proc chpl_typeMoveWarning(param name: string, param mod: string,
+                            param newmod: string = "CTypes") {
+    compilerWarning("type '" + name + "' has moved from '" + mod +
+                    "' to '" + newmod + "'; please update your " +
+                    "'use'/'import' statements accordingly.", errorDepth=2);
+  }
+
+  /* The type corresponding to a C float */
+  extern type c_float = real(32);
+
+  /* The type corresponding to a C double */
+  extern type c_double = real(64);
+
+  // Former CPtr contents start here
+
+  /* A Chapel type alias for 'void*' in C. */
+  extern type c_void_ptr = chpl__c_void_ptr;
+
+
+  /* A Chapel version of a C NULL pointer. */
```

Comment:
Could you add a Future Work note to the PR message about potentially deprecating c_nil and is_c_nil ? (It seems clear to me that we should deprecate them but I don't know if more discussion is needed).

Reply:
Done.  I basically clarified that this is not meant to be the final/blessed version of CTypes, just the one that brings the various moving parts together into one module.  I wanted to keep the review load easy... ;)